pkg/jobs/jobs.go

```diff
@@ -0,0 +1,128 @@
+package jobs
+
+import (
+	"bytes"
+	"encoding/gob"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+const (
+	// Queued state
+	Queued State = "queued"
+	// Running state
+	Running = "running"
+	// Errored state
+	Errored = "errored"
+)
+
+const (
+	// JSONEncoding is a JSON encoding message type
+	JSONEncoding = "json"
+	// GobEncoding is a Gob encoding message type
+	GobEncoding = "gob"
+)
+
+type (
+	// Queue interface is used to represent a asynchronous queue of jobs from
+	// which it is possible to enqueue and consume jobs.
+	Queue interface {
+		Enqueue(*Job) error
+		Consume() (bool, *Job, error)
```

Comment:
To alert the consumer that the queue is closed. Might not be really idiomatic though.